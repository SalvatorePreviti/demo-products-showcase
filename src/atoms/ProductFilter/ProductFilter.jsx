import React from 'react'
import PropTypes from 'prop-types'
import styles from './ProductFilter.scss'

function ProductFilter(props) {
  const { value, onChange } = props
  return (
    <div className={styles.ProductFilter}>
      <input type="text" value={value} onChange={event => onChange && onChange(event.target.value)} />
    </div>
  )
}

ProductFilter.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func
}

export default ProductFilter
