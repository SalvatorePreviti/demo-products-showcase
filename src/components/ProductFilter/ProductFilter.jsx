import React from 'react'
import PropTypes from 'prop-types'

function ProductFilter(props) {
  const { value, onChange } = props
  return (
    <div>
      <input type="text" value={value} onChange={onChange} />
    </div>
  )
}

ProductFilter.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func
}

export default ProductFilter
