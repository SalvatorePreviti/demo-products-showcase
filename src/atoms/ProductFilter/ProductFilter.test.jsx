import React from 'react'
import { shallow } from 'enzyme'

import ProductFilter from './ProductFilter'

describe('ProductItem', () => {
  let root

  beforeAll(() => {
    root = shallow(<ProductFilter value="custom filter" />)
  })

  it('snapshots', () => {
    expect(root).toMatchSnapshot()
  })

  it('renders the input box correctly', () => {
    expect(root.find('input').props()).toMatchObject({})
  })
})
