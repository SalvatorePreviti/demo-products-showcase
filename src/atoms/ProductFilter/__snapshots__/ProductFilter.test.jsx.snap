// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem snapshots 1`] = `
<div>
  <input
    type="text"
    value="custom filter"
  />
</div>
`;
